pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -42,6 +42,13 @@ const (
 	// It is used for multi-cluster scenario, and with nodePort type gateway service.
 	// TODO: move to API
 	NodeSelectorAnnotation = "traffic.istio.io/nodeSelector"
+
+	// LBExternalAddressesAnnotation overrides the external address value with a LoadBalancer type service.
```

Comment:
Somewhere in the doc/comment, make sure we emphasize this overwrites and does not append.

Reply:
Yes, add the same annotations and documentation to [istio/api](https://github.com/istio/api) once this PR is merged.
https://github.com/istio/api/blob/master/annotation/annotations.yaml

Reply:
https://github.com/istio/api/pull/2858